### ISSUE TITLE
charge assignment from file

### ISSUE THREAD

**Issue by mj-harvey** [openbabel/openbabel#157]
Hi,

I'd like to be able to assign arbitrary charges to atoms, from a file containing a list of atomname charge pairs, for example

```
babel --partialcharges fromfile [charges.txt] -i mol2 input.mol2 -o mol2 output.mol2
```

I'm writing a plugin to do this, since there doesn't seem to be any way to do it currently.
Any recommendations about how best to pass the filename argument to the charge method?

Matt

**Comment by ghutchis** [comment]
This would probably require changes to the OBChargeModel class in includes/ and src/chargemodel.cpp as well as the op/partialcharges.cpp file to parse the filename. I'd suggest contacting the openbabel-devel mailing list and I can provide more details if needed.

**Comment by ghutchis** [comment]
I'd be happy to draft something for the OBChargeModel class. It's something I considered previously to allow parameterization, e.g. of the EEM methods.

**Comment by mj-harvey** [comment]
Hi,

That'd be very much appreciated, thank you!

Matt

**Comment by mj-harvey** [comment]
See my pull request for a proposal for this. The charge mapping file is passed as a suffix to the partial charge method name:

```babel --partialcharge fromfile:mapping.txt ... ```

The alternative seemed to be to add a top-level GENOPTIONS flag, and I wanted to avoid polluting that namespace unnecessarily.

Comments?

**Comment by mj-harvey** [comment]
I should say, I've another PC-related method that I'd like to merge, which will also require an input argument (this scheme will work there too).

**Comment by ghutchis** [comment]
This is solved by #168 and #209.
